config/v1/types_cluster_version.go

```diff
@@ -215,6 +222,80 @@ type UpdateHistory struct {
 // ClusterID is string RFC4122 uuid.
 type ClusterID string
 
+// ClusterVersionCapability enumerates optional, core cluster components.
+type ClusterVersionCapability string
+
+const (
+	// ClusterVersionCapabilityOpenShiftSamples manages the sample
+	// image streams and templates stored in the openshift
+	// namespace, and any registry credentials, stored as a secret,
+	// needed for the image streams to import the images they
+	// reference.
+	ClusterVersionCapabilityOpenShiftSamples ClusterVersionCapability = "openshift-samples"
+)
+
+// ClusterVersionCapabilitySet defines sets of cluster version capabilities.
+type ClusterVersionCapabilitySet string
+
+const (
+	// ClusterVersionCapabilitySetNone is an empty set containing no capabilities.
+	ClusterVersionCapabilitySetNone ClusterVersionCapabilitySet = "None"
+
+	// ClusterVersionCapabilitySet4_11 is the recommended set of
+	// optional capabilities for the 4.11 version of OpenShift.
+	// This list will remain the same no matter which version of
+	// OpenShift is installed.
+	ClusterVersionCapabilitySet4_11 ClusterVersionCapabilitySet = "v4.11"
+
+	// ClusterVersionCapabilitySetCurrent is the recommended set of
+	// optional capabilities for the current version of OpenShift.
+	ClusterVersionCapabilitySetCurrent ClusterVersionCapabilitySet = "vCurrent"
+)
+
+// ClusterVersionCapabilitySets defines sets of cluster version capabilities.
+var ClusterVersionCapabilitySets = map[ClusterVersionCapabilitySet][]ClusterVersionCapability{
+	ClusterVersionCapabilitySetNone: {},
+	ClusterVersionCapabilitySet4_11: {
+		ClusterVersionCapabilityOpenShiftSamples,
+	},
+	ClusterVersionCapabilitySetCurrent: {
+		ClusterVersionCapabilityOpenShiftSamples,
+	},
+}
+
+// ClusterVersionCapabilitiesSpec selects the managed set of
+// optional, core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesSpec struct {
+	// baselineCapabilitySet selects an initial set of
+	// capabilities, which can be extended via
+	// additionalEnabledCapabilities.
+	// +kubebuilder:default:="vCurrent"
+	BaselineCapabilitySet ClusterVersionCapabilitySet `json:"baselineCapabilitySet,omitempty"`
+
+	// additionalEnabledCapabilities extends the set of managed
+	// capabilities beyond the baseline defined in
+	// baselineCapabilitySet.
+	// +listType=atomic
+	// +optional
+	AdditionalEnabledCapabilities []ClusterVersionCapability `json:"additionalEnabledCapabilities,omitempty"`
+}
+
+// ClusterVersionCapabilitiesStatus describes the state of optional,
+// core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesStatus struct {
+	// enabledCapabilities lists all the capabilities that are currently managed.
+	// +listType=atomic
+	// +optional
+	EnabledCapabilities []string `json:"enabledCapabilities,omitempty"`
```

Comment:
@deads2k, responding to your [`enabledCapabilities` suggestion][1], I'd floated `enabled`, because:

```yaml
status:
  capabilities:
    enabledCapabilities: ...
```

seems like redundant, unnecessary context.  You'd cited [`FeatureGateSelection.FeatureSet`][2] as precedent, but that's consumed with `inline`:

```go
type FeatureGateSpec struct {
	FeatureGateSelection `json:",inline"`
}
```

And I agree that:

```yaml
spec:
  set: ...
```

would be too short on context.  The structures in this PR will not be `inline`.  Do you have precedent you can site for parent-name repetition in the absence of `inline`?  Hunting in the API with [this script](https://github.com/openshift/api/files/8006815/parent-echo.py.txt), I'm not seeing anything I'm particularly excited to mimic:

```console
$ ./parent-echo.py 
./config/v1/0000_10_config-operator_01_infrastructure.crd.yaml
  'platformSpec' includes its parent 'spec'
  'platformStatus' includes its parent 'status'
./imageregistry/v1/00_imageregistry.crd.yaml
  'nodeAffinity' includes its parent 'affinity'
  'podAffinity' includes its parent 'affinity'
  'podAntiAffinity' includes its parent 'affinity'
  'noProxy' includes its parent 'proxy'
./imageregistry/v1/01_imagepruner.crd.yaml
  'nodeAffinity' includes its parent 'affinity'
  'podAffinity' includes its parent 'affinity'
  'podAntiAffinity' includes its parent 'affinity'
./operator/v1/0000_12_etcd-operator_01_config.crd.yaml
  'nodeStatuses' includes its parent 'status'
./operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml
  'nodeStatuses' includes its parent 'status'
./operator/v1/0000_25_kube-controller-manager-operator_01_config.crd.yaml
  'nodeStatuses' includes its parent 'status'
./operator/v1/0000_25_kube-scheduler-operator_01_config.crd.yaml
  'nodeStatuses' includes its parent 'status'
./quota/v1/0000_03_quota-openshift_01_clusterresourcequota.crd.yaml
  'matchLabels' includes its parent 'labels'
./machine/v1beta1/0000_10_machine.crd.yaml
  'providerSpec' includes its parent 'spec'
  'providerStatus' includes its parent 'status'
./machine/v1beta1/0000_10_machineset.crd.yaml
  'providerSpec' includes its parent 'spec'
./networkoperator/v1/001-egressrouter.crd.yaml
  'redirectRules' includes its parent 'redirect'
```

[1]: https://github.com/openshift/api/pull/1106#discussion_r799819632
[2]: https://github.com/openshift/api/blob/31ffd77a8f025c73b5d19b390af353f9775e7ef1/config/v1/types_feature.go#L53-L58